src/main/kotlin/com/sourcegraph/cody/chat/ExportChatsBackgroundable.kt

```diff
@@ -0,0 +1,62 @@
+package com.sourcegraph.cody.chat
+
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.ChatHistoryResponse
+import com.sourcegraph.cody.agent.protocol.IdParam
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.cody.history.HistoryService
+import java.util.concurrent.TimeUnit
+
+class ExportChatsBackgroundable(
+    project: Project,
+    private val onSuccess: (ChatHistoryResponse) -> Unit,
+    private val onFinished: () -> Unit
+) : Task.Backgroundable(project, /* title = */ "Exporting chats...", /* canBeCancelled = */ true) {
+
+  override fun run(indicator: ProgressIndicator) {
+    val accountId = CodyAuthenticationManager.instance.getActiveAccount(project)?.id
+    val chats =
+        HistoryService.getInstance(project)
+            .state
+            .chats
+            .filter { it.accountId == accountId }
+            .filter { it.messages.isNotEmpty() }
+
+    chats.forEachIndexed { index, chatState ->
+      AgentChatSessionService.getInstance(project).getOrCreateFromState(chatState)
+      indicator.fraction = ((index + 1.0) / (chats.size + 1.0))
+      if (indicator.isCanceled) {
+        return
+      }
+    }
+
+    AgentChatSession.createNew(project) { connectionId ->
+      CodyAgentService.withAgent(project) { agent ->
+        val result: ChatHistoryResponse? =
+            agent.server
+                .chatExport(IdParam(connectionId))
+                .completeOnTimeout(null, 15, TimeUnit.SECONDS)
+                .get()
+        if (result != null) {
+          onSuccess.invoke(result)
+        } else {
+          throw Error("Request timed out") // todo: handle it
```

Comment:
@pkukielka, this is a scenario when we time out. Do you have any suggestions how we should handle that? 🙏

Reply:
We are timing out because of `.completeOnTimeout(null, 15, TimeUnit.SECONDS)` or something else?

Reply:
I just tried with a chat that has **over 5000 messages** (the number of assistant messages) - it fails 🔴 

```
2024-03-21 16:42:32,131 [ 342766]   WARN - #c.i.d.PerformanceWatcherImpl - UI was frozen for 5191ms, details saved to /Users/mkondratek/IdeaProjects/jetbrains/build/idea-sandbox/system/log/threadDumps-freeze-20240321-164231-IC-221.5080.210-5sec
java.lang.Throwable: Invocation timed out (1sec)
	at java.desktop/sun.lwawt.macosx.LWCToolkit.invokeAndWait(LWCToolkit.java:818)
	at java.desktop/sun.lwawt.macosx.LWCToolkit.invokeAndWait(LWCToolkit.java:664)
	at java.desktop/sun.lwawt.macosx.CAccessibility.invokeAndWait(CAccessibility.java:114)
	at java.desktop/sun.lwawt.macosx.CAccessibility.invokeAndWait(CAccessibility.java:108)
	at java.desktop/sun.lwawt.macosx.CAccessibility.getFocusOwner(CAccessibility.java:568)
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
java.lang.Throwable: Invocation timed out (1sec)
	at java.desktop/sun.lwawt.macosx.LWCToolkit.invokeAndWait(LWCToolkit.java:818)
	at java.desktop/sun.lwawt.macosx.LWCToolkit.invokeAndWait(LWCToolkit.java:664)
	at java.desktop/sun.lwawt.macosx.CAccessibility.invokeAndWait(CAccessibility.java:114)
	at java.desktop/sun.lwawt.macosx.CAccessibility.invokeAndWait(CAccessibility.java:108)
	at java.desktop/sun.lwawt.macosx.CAccessibility.getFocusOwner(CAccessibility.java:568)
Exception in thread "HttpClient-1-SelectorManager" java.lang.OutOfMemoryError: Java heap space
```

I tried with ~1500 messages (the number of assistant messages) - it works (I was able to export) but IntelliJ complains about running low on memory.

I tried with 1000 messages (the number of assistant messages) - it works - no complaints.